kqueen_ui/blueprints/ui/views.py

```diff
@@ -618,14 +617,23 @@ def handle(self):
                 render_template('ui/cluster_create.html', form=form)
 
             owner_ref = 'User:{}'.format(session['user']['id'])
+            provisioner_id = form.provisioner.data
             cluster_kw = {
                 'name': form.name.data,
                 'state': app.config['CLUSTER_PROVISIONING_STATE'],
-                'provisioner': 'Provisioner:{}'.format(form.provisioner.data),
+                'provisioner': 'Provisioner:{}'.format(provisioner_id),
                 'created_at': datetime.utcnow(),
                 'metadata': metadata,
                 'owner': owner_ref
             }
+
+            network_policy = form['network_policy' + '__' + provisioner_id].data
+            if network_policy == 'calico':
+                cluster_kw['network_policy'] = {
+                    'provider': 'CALICO',
+                    'enabled': True
+                }
```

Comment:
If you get data from select form, then it will always have a value, so we can not use just `if network_policy`. And I don't see a reason to assign this to metadata, as the backend expects the `network_policy` parameter in the arguments.

Reply:
The backend should expect simple value provided directly by the form though, so I would remove lines 630 - 635 entirely and just make do with what is rendered by default according to parameter schema from backend. If its not sufficient, then there are changes to be done in the backend, but not here. This works in a generic manner and it shouldn't be difficult to keep it that way